### PR TITLE
Update Juju dependencies to recent versions

### PR DESCRIPTION
**Pull request by taurus-forever**
We were testing using old Juju for no reason. The modern versions should work well.
While we are here, updating library data_platform_libs.s3.

### CODE REVIEW

File: lib/charms/data_platform_libs/v0/s3.py [hunk 1]
```diff
@@ -250,6 +250,41 @@ def _load_relation_data(self, raw_relation_data: dict) -> dict:
...
+    #         key for key in old_data.keys() & new_data.keys() if old_data[key] != new_data[key]
+    #     }
+
+    #     # TODO: evaluate the possibility of losing the diff if some error
+    #     # happens in the charm before the diff is completely checked (DPE-412).
+    #     # Convert the new_data to a serializable format and save it for a next diff check.
+    #     event.relation.data[self.local_app].update({"data": json.dumps(new_data)})
+
+    #     # Return the diff with all possible changes.
+    #     return Diff(added, changed, deleted)
```
Comment (deusebio) [inline lib/charms/data_platform_libs/v0/s3.py:286]: Should we just remove this?

**Review by deusebio — APPROVED** [review]
All good!

File: lib/charms/data_platform_libs/v0/s3.py (hunk 1 above)
Comment (taurus-forever) [reply lib/charms/data_platform_libs/v0/s3.py:286]: Sure, go for it.
It is a library coming from [another](https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/s3.py) repository (removal is not applicable to this PR).

Comment (deusebio) [reply lib/charms/data_platform_libs/v0/s3.py:286]: Yes, right! Sorry 

The charmlib update are a bit confusing in this respect (what we should review and what we should not). With python dependensies is easier...whatever is in the reqs.txt we don't mind :joy:

**Review by delgod — APPROVED** [review]

**Review by welpaolo — APPROVED** [review]

**Pull request merged**